console/console-init/ui/src/modules/device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -0,0 +1,388 @@
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Form,
+  FormGroup,
+  TextInput,
+  DropdownPosition,
+  InputGroup,
+  InputGroupText,
+  TextContent,
+  Divider,
+  Grid,
+  GridItem
+} from "@patternfly/react-core";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { CalendarAltIcon, PlusCircleIcon } from "@patternfly/react-icons";
+import { DropdownWithToggle } from "components";
+import { ISelectOption, uniqueId, findIndexByProperty } from "utils";
+import {
+  IDeviceFilterCriteria,
+  DeviceFilterCriteria,
+  operator
+} from "modules/device";
+
+const styles = StyleSheet.create({
+  time_input_box: {
+    padding: 20,
+    marginRight: 10
+  },
+  dropdown_align: { display: "flex", marginRight: 10 },
+  dropdown_toggle_align: { flex: "1" }
+});
+
+const setInitialFilter = () => {
+  let filter: IDeviceFilter = {
+    deviceId: "",
+    deviceType: "gateway",
+    status: "enabled",
+    filterCriteria: [],
+    addedDate: {
+      startDate: "",
+      endDate: ""
+    },
+    lastSeen: {
+      endTime: {
+        form: "hr",
+        time: ""
+      },
+      startTime: {
+        form: "min",
+        time: ""
+      }
+    }
+  };
+  return filter;
+};
+
+interface ITimeOption {
+  time: string;
+  form: string;
+}
+export interface IDeviceFilter {
+  deviceId: string;
+  deviceType: string;
+  status: string;
+  lastSeen: {
+    startTime: ITimeOption;
+    endTime: ITimeOption;
+  };
+  addedDate: {
+    startDate: string;
+    endDate: string;
+  };
+  filterCriteria: IDeviceFilterCriteria[];
+}
+export interface IDeviceFilterProps {
+  filter?: IDeviceFilter;
+  setFilter?: (filter: IDeviceFilter) => void;
+}
+const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = (
+  {
+    // filter,
+    // setFilter
+  }
+) => {
+  const [filter, setFilter] = useState<IDeviceFilter>(setInitialFilter());
+  const typeOptions: ISelectOption[] = [
+    {
+      key: "direct",
+      value: "direct",
+      label: "Directly connected",
+      isDisabled: false
+    },
+    {
+      key: "gateway",
+      value: "gateway",
+      label: "Using gateways",
+      isDisabled: false
+    }
+  ];
+  const statusOptions: ISelectOption[] = [
+    {
+      key: "enabled",
+      value: "enabled",
+      label: "Enabled",
+      isDisabled: false
+    },
+    {
+      key: "disabled",
+      value: "disabled",
+      label: "Disabled",
+      isDisabled: false
+    }
+  ];
+  const timeOptions: ISelectOption[] = [
+    {
+      key: "hr",
+      value: "hr",
+      label: "hr",
+      isDisabled: false
+    },
+    {
+      key: "min",
+      value: "min",
+      label: "min",
+      isDisabled: false
+    },
+    {
+      key: "sec",
+      value: "sec",
+      label: "sec",
+      isDisabled: false
+    }
+  ];
+
+  const setTime = (value: string, isStart: boolean, isFormat: boolean) => {
```

Comment:
This function logic cane be more simple, there are so much if else conditions.
Pass property name instead of isStart. 
  const setTime = (value: string, propertyName: string, isFormat: boolean) => {
    const filterObj = { ...filter };
    if (propertyName) {
      let timeObj = filter.lastSeen[propertyName];
      if (isFormat) {
        timeObj = { form: value, time: timeObj.time };
      } else {
        timeObj = { form: timeObj.form, time: value };
      }
      filterObj.lastSeen[propertyName] = timeObj;
     setFilter(filterObj); 
    }
  };
Note: you can pass propertyName as startTime and endTime and it will work as expected.

setTime(value, "startTime", false);
setTime(value, "endTime", false);